src/CiviEntityStorage.php

```diff
@@ -258,7 +258,13 @@ public function countFieldData($storage_definition, $as_bool = FALSE) {
    * {@inheritdoc}
    */
   public function hasData() {
-    return $this->getCiviCrmApi()->getCount($this->entityTypeId, []) > 0;
+    if (!$this->civicrmApi) {
+      $this->civicrmApi = $this->getCiviCrmApi();
+    }
+    if (substr($this->entityTypeId, 0, 8) == 'civicrm_') {
+      return $this->civicrmApi->getCount(substr($this->entityTypeId, 8)) > 0;
+     }
+    return $this->civicrmApi->getCount($this->entityTypeId) > 0;
```

Comment:
However, I think this change might be hiding a problem elsewhere, or maybe missing a utility function. Somewhere there's a list of civi -> drupal entity ids, or a utility to do that lookup, and that should be used here. Lemme poke around the code....

Reply:
Alright, I've got it! This method should just be:

```
  public function hasData() {
    return $this->civicrmApi->getCount($this->entityType->get('civicrm_entity')) > 0;
  }
```

Basically, before it was trying to use the Drupal entity type id for the CiviCRM API, and this changes it to use the CiviCRM entity type id.

Reply:
Thanks David! Ah ok - let me verify/make that edit

Reply:
Unfortunately that produces: `Error: Call to a member function getCount() on null in Drupal\civicrm_entity\CiviEntityStorage->hasData() (line 261 `

I did `drush cr` twice after making the edit

I still need to do this (in order to not get the fatal error on the uninstall screen):
```

    if (!$this->civicrmApi) {
      $this->civicrmApi = $this->getCiviCrmApi();
    }
    return $this->civicrmApi->getCount($this->entityType->get('civicrm_entity')) > 0;


```

Reply:
Hm. I'm not sure how this bit helps:

```
if (!$this->civicrmApi) {
  $this->civicrmApi = $this->getCiviCrmApi();
}
```

There is no `$this->getCiviCrmApi()` method on `CiviEntityStorage` AND `$this->civicrmApi` should never be NULL because it's required by the class constructor...

Reply:
I just tested without that bit of code and the "Uninstall" page is loading for me... Not sure what we're doing differently in our environments?

Reply:
Wicked! I've seen this on two sites/two servers thus far. Let me try again (going backwards on d8.semper-it.com):

**1. I can get to Uninstall screen:**
![image](https://user-images.githubusercontent.com/5340555/64981852-2e8b4680-d87a-11e9-83ae-fda8025b0db0.png)

**2. Commenting out these lines:**
![image](https://user-images.githubusercontent.com/5340555/64982050-89bd3900-d87a-11e9-8773-5eeaeb11653e.png)

**3. Reload Uninstall Screen:**
`Error: Call to a member function getCount() on null in Drupal\civicrm_entity\CiviEntityStorage->hasData() (line 264 of /var/www/drupal/web/modules/contrib/civicrm_entity/src/CiviEntityStorage.php) #0 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentUninstallValidator.php(56): Drupal\civicrm_entity\CiviEntityStorage->hasData() `

If you can think of anythings else I can try - let me know.

Reply:
Ok mystery solved -> `$this->getCiviCrmApi` got introduced in https://github.com/eileenmcnaughton/civicrm_entity/pull/182 - which David does not use yet.